### PR TITLE
Deprecate fancy printing and internal methods

### PR DESCRIPTION
**Pull request by theotherjimmy**

### CODE REVIEW

File: mbed_lstools/lstools_base.py [hunk 1]
```diff
@@ -308,23 +324,27 @@ def get_string(self, border=False, header=True, padding_width=1, sortby='platfor
 
     # Private functions supporting API
 
+    @deprecated("Using internal functions is deprecated.")
```
Comment (bridadan) [inline mbed_lstools/lstools_base.py]: For this particular deprecated message I think I'd give a bit more of a warning/reason.

"This function will be removed from the public API in future releases. Please use 'list_mbeds' instead."

File: test/listing.py [hunk 2]
```diff
@@ -66,20 +66,6 @@ def test_list_mbeds_ext_mandatory_fields_exist(self):
...
-        for p in platforms:
-            self.assertIs(type(p), str)
-
-    def test_list_platforms_ext(self):
-        platforms = self.mbeds.list_platforms_ext()
-        self.assertIs(type(platforms), dict)
-        for p in platforms:
-            self.assertIs(type(p), str)
-            self.assertIs(type(platforms[p]), int)
-            self.assertTrue(platforms[p] >= 0)
```
Comment (bridadan) [inline test/listing.py]: Even if functions are deprecated, should we still test them? Seems like when they're removed it'd be appropriate to stop testing them.

If you feel strongly the other way then no worries, just my initial reaction.

File: test/mbedls_toolsbase.py [hunk 3]
```diff
@@ -63,10 +63,6 @@ def test_list_mbeds(self):
         self.assertEqual(to_check[0]['target_id'], "0241BEEFDEAD")
         self.assertEqual(to_check[0]['platform_name'], 'foo_target')
 
-    def test_list_manufacture_ids(self):
-        table_str = self.base.list_manufacture_ids()
-        self.assertTrue(isinstance(table_str, basestring))
```
Comment (bridadan) [inline test/mbedls_toolsbase.py]: Ditto here about testing

File: mbed_lstools/lstools_base.py [hunk 4]
```diff
@@ -128,6 +139,7 @@ def mock_manufacture_id(self, mid, platform_name, oper='+'):
         else:
             raise ValueError("oper can only be [+-]")
 
+    @deprecated("Fancy target listing is out of scope of mbedls.")
```
[the remark this answers is not in the document]
Comment (theotherjimmy) [reply mbed_lstools/lstools_base.py]: That is snarky, but I would not say that it's less technical. I'm going to change it.

File: test/listing.py (hunk 2 above)
Comment (theotherjimmy) [reply test/listing.py]: I guess. I don't feel strongly either way.

**Comment by theotherjimmy** [comment]
@bridadan Ready for review.

**Pull request merged**